src/gwt/src/org/rstudio/studio/client/workbench/views/source/editors/text/ace/AceEditorBackgroundLinkHighlighter.java

```diff
@@ -0,0 +1,681 @@
+      Element markerEl = el.hasClassName(id)
+            ? el
+            : DomUtils.getFirstElementWithClassName(el, id);
+      if (markerEl == null)
+         return;
+      
+      if (activeHighlightMarkerEl_ != null && activeHighlightMarkerEl_ != markerEl)
+      {
+         activeHighlightMarkerEl_.addClassName(RES.styles().highlight());
+         activeHighlightMarkerEl_.removeClassName(RES.styles().hover());
+      }
+      
+      markerEl.removeClassName(RES.styles().highlight());
+      markerEl.addClassName(RES.styles().hover());
+      activeHighlightMarkerEl_ = markerEl;
+   }
+   
+   private void endDetectClickTarget()
+   {
+      if (activeHighlightMarkerEl_ == null)
+         return;
+      
+      // restore highlight styles
+      activeHighlightMarkerEl_.addClassName(RES.styles().highlight());
+      activeHighlightMarkerEl_.removeClassName(RES.styles().hover());
+      
+      // unset active el
+      activeHighlightMarkerEl_ = null;
+   }
+   
+   private void clearMarkers(final Range range)
+   {
+      for (int row = range.getStart().getRow();
+           row <= range.getEnd().getRow();
+           row++)
+      {
+         if (!activeMarkers_.containsKey(row))
+            continue;
+         
+         // clear markers that are included within this range
+         List<MarkerRegistration> markers = activeMarkers_.get(row);
+         List<MarkerRegistration> filtered = ListUtil.filter(markers, new FilterPredicate<MarkerRegistration>()
+         {
+            @Override
+            public boolean test(MarkerRegistration marker)
+            {
+               if (range.contains(marker.getRange()))
+               {
+                  marker.detach();
+                  return false;
+               }
+               
+               return true;
+            }
+         });
+         
+         // update active markers for this row
+         activeMarkers_.put(row, filtered);
+      }
+   }
+   
+   private void navigateToUrl(String url)
+   {
+      // allow web links starting with 'www'
+      if (url.startsWith("www."))
+         url = "http://" + url;
+      
+      // attempt to open web links in a new window
+      Pattern reWebLink = Pattern.create("^https?://");
+      if (reWebLink.test(url))
+      {
+         globalDisplay_.openWindow(url);
+         return;
+      }
+      
+      // handle testthat links
+      Pattern reSrcRef = Pattern.create("@[^#]+#\\d+");
+      if (reSrcRef.test(url))
+         return;
+      
+      // treat other URLs as paths to files on the server
+      final String finalUrl = url;
+      server_.stat(finalUrl, new ServerRequestCallback<FileSystemItem>()
+      {
+         @Override
+         public void onResponseReceived(FileSystemItem file)
+         {
+            // inform user non-obtrusively if no file found
+            if (file == null || !file.exists())
+            {
+               globalDisplay_.showWarningBar(false, "No file at path '" + finalUrl + "'.");
```

Comment:
Given that we don't have access to the TextEditingTarget.Display here (we only have the AceEditor class, which doesn't know about its parent widgets) I'll stick with the slightly more aggressive prompt.